src/model_repository_manager.h

```diff
@@ -260,65 +267,119 @@ class ModelRepositoryManager {
+  /// \param new_graph The new graph to be cleared and copied into
+  void CopyDependencyGraph(DependencyGraph* new_graph) const;
+
+  /// Helper function to re-map dependency graph nodes pointers from old nodes
+  /// to new nodes.
+  /// The function search for matching pointers in the new_nodes first, if not
+  /// found, then the matching pointer must be in the ref_nodes.
+  /// \param ref_nodes The reference nodes for pointers not in new_nodes
+  /// \param new_nodes The new nodes to be re-mapped
+  void ReMapDependencyGraphPointers(
+      const std::unordered_map<std::string, std::unique_ptr<DependencyNode>>*
+          ref_nodes,
+      std::unordered_map<std::string, std::unique_ptr<DependencyNode>>*
+          new_nodes) const;
 
   /// Helper function to uncheck the nodes because the model that they depends
   /// on has changed. The unchecked nodes will be validated again.
   /// The function will be call recursively to uncheck all downstreams.
   /// \param downstreams The nodes to be unchecked.
   /// \param updated_nodes Return the nodes that have been unchecked
-  void UncheckDownstream(NodeSet* downstreams, NodeSet* updated_nodes);
+  void UncheckDownstream(NodeSet* downstreams, NodeSet* updated_nodes) const;
 
   /// Helper function to construct the edges between nodes in dependency graph.
+  /// \param graph The dependency graph.
   /// \param updated_node The node that is newly added or modified.
   /// \return True if the node represents an ensemble model. False otherwise.
-  bool ConnectDependencyGraph(DependencyNode* updated_node);
+  bool ConnectDependencyGraph(
+      DependencyGraph* graph, DependencyNode* updated_node) const;
 
   /// Get the model info for a named model.
+  /// \param model_infos Map of model names to its infomation.
   /// \param name The model name.
   /// \param model_info Returns the model information.
   /// \return OK if found, NOT_FOUND otherwise.
-  Status GetModelInfo(const std::string& name, ModelInfo** model_info);
+  Status GetModelInfo(
+      const ModelInfoMap& model_infos, const std::string& name,
+      ModelInfo** model_info) const;
+
+  /// Helper function to deep copy the model infos
+  /// \param new_infos The new model infos to be cleared and copied into
+  void CopyModelInfos(ModelInfoMap* new_infos) const;
+
+  /// Check if a dependency node (model) is currently loading/unloading
+  /// \param dependency_node The dependency node to check
+  /// \return OK if NOT in transit, INVALID_ARG if in transit
+  Status InTransit(const DependencyNode* dependency_node) const;
+
+  /// Mark the nodes (model) in dependency graph to currently loading/unloading
+  /// \param graph The dependency graph
+  /// \param names The name of models to be in transition
+  /// \param transition True for loading, False for unloading
+  void UpdateTransition(
+      const DependencyGraph& graph, const std::set<std::string>& names,
+      bool transition) const;
```

Comment:
Yes. The `graph` should not be a `const` reference. Thanks for catching.

Reply:
Use "pointer to" (`*`) instead of reference (`&`) if you are going to modify the object, two ways are essentially the same but we often use * to indicate that the parameter is output parameter

Reply:
Yes. Updated.